File: Kotlin/Two_Sum.kt
import java.util.Arrays

fun twoSum(nums: IntArray, target: Int): IntArray {
    val lookingFor = HashMap<Int, Int>();
    for (i in nums.indices) {
        if (lookingFor.containsKey(nums[i])) return intArrayOf(lookingFor.get(nums[i])!!, i);
        lookingFor.put(target - nums[i], i);
    }

    return intArrayOf(0, 0);
}

fun main(args: Array<String>) {
    println(Arrays.toString(twoSum(intArrayOf(2, 11, 15, 7), 9)));
}

File: Go/Two_Sum.go
func twoSum(nums []int, target int) []int {
	lookingFor := map[int]int{}
	for i, v := range nums {
		if val, ok := lookingFor[v]; ok {
			return []int{val, i}
		}
		lookingFor[target-v] = i
	}
	return []int{-1, -1}
}